services/librariesio/librariesio-dependencies.service.js

```diff
@@ -1,134 +1,137 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
-const { checkErrorResponse } = require('../../lib/error-helper')
+const Joi = require('joi')
+const { BaseJsonService } = require('..')
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class LibrariesioDependencies extends LegacyService {
+function transform({ dependencies }) {
+  return {
+    deprecatedCount: dependencies.filter(dep => dep.deprecated).length,
+    outdatedCount: dependencies.filter(dep => dep.outdated).length,
+  }
+}
+
+function renderDependenciesBadge({ deprecatedCount, outdatedCount }) {
+  if (deprecatedCount > 0) {
+    // Deprecated dependencies are really bad
+    return {
+      message: `${deprecatedCount} deprecated`,
+      color: 'red',
+    }
+  } else if (outdatedCount > 0) {
+    // Out of date dependencies are pretty bad
+    return {
+      message: `${outdatedCount} out of date`,
+      color: 'orange',
+    }
+  } else {
+    return {
+      message: 'up to date',
+      color: 'brightgreen',
+    }
+  }
+}
+
+const schema = Joi.object({
+  dependencies: Joi.array().default([]),
```

Comment:
Oops, forgot to update this!

Reply:
I'm good to approve this PR once you're set with the schema 👍

Reply:
Added!